src/devices/src/virtio/net/device.rs

```diff
@@ -217,9 +218,31 @@ impl Net {
         self.guest_mac.as_ref()
     }
 
-    /// Provides a mutable reference to the `MmdsNetworkStack`.
-    pub fn mmds_ns_mut(&mut self) -> Option<&mut MmdsNetworkStack> {
-        self.mmds_ns.as_mut()
+    /// Provides the host IFACE name of this net device.
+    pub fn iface_name(&self) -> String {
```

Comment:
Instead of adding these many member fns, I would look at defining a trait like `ConfigDump` and implement it for all the objects we want to export the configuration as JSON. We have a similar pattern with the `Persist` trait.

Reply:
This is what I initially went for, but it evolved into current form because:

Current Firecracker design/architecture has JSON/API configuration/description structs defined as a submodule of `vmm` crate as we considered them to be vmm-specific. The idea was to keep devices more generic and not tie them to the configuration model and methods.
So `devices` crate should not depend on these _configuration_ struct definitions.

We could define the trait in the config, then implement the trait for all devices, but I lean towards implementing `std::convert::From` to get _&device -> config_ instead of a custom trait in the same place with same functionality and usage.

The added member fns are simple immutable getters, I'm not aware of any disadvantage in having them.

Reply:
Gotcha. The idea was to reduce LOC by removing the getters.

Reply:
I like `From` and the `fn`s better than a custom trait.